### PR TITLE
Closes CPPL-275

### PR DESCRIPTION
**Pull request by iodream**
Adds language changing support and sections in Settings as well as the Login Page.
Also adds UI language as another parameter which is saved in Settings.json

Translation usage:
Translation files need to be present in the same directory as the executable. QT translation files are optional but need to be copied into the same location to have full effect. 
 
To create new translations you need to copy one of the *.ts files, rename it according to the language, add them to Cmake, modify the language parameter at the top of the file and then import it into QT linguist. 

To add translation strings: use tr() around your text, rebuild the client and then translate the new strings inside qt linguist. 

Known bugs: can't read UI Language when logging in. 
Can't safely inform user of error if translation files are missing. Only in debug for now. 

Screenshot_20220502_190313

Screenshot_20220502_190341

### CODE REVIEW

File: src/Client/Controller/Controller.cpp [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
 #include "Models/LoginModel.h"
 #include "Models/List/GetListsModel.h"
 
 #include "Net/Constants.h"
 
-Controller::Controller()
+Controller::Controller(QApplication* a)
+	:m_app(a)
 {
+	m_trans = std::make_shared<QTranslator>();
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/Controller.cpp]: Don't abbreviate words without good reason.

File: src/Client/Controller/Controller.cpp [hunk 2]
```diff
@@ -97,6 +115,13 @@ void Controller::InitLoginPageController()
 		&LoginPageController::ChangePage,
 		this,
 		&Controller::OnChangePage);
+	connect(
+		m_login_page_controller.get(),
+		&LoginPageController::LangChanged,
+		this,
+		&Controller::OnLangChanged);
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/Controller.cpp]: "lang" abbreviation is obvious in this context, but why not just "language"?

File: src/Client/Controller/Controller.cpp [hunk 3]
```diff
@@ -135,12 +160,18 @@ void Controller::InitMainPageController()
...
 }
 
 bool Controller::AskUser(const QString& title, const QString& text)
 {
-	return QMessageBox::Yes == QMessageBox::question(nullptr, QString("Retry?"),
-													 QString("No connection to server. Retry?"));
+	return QMessageBox::Yes == QMessageBox::question(nullptr, tr("Retry?"),
+											tr("No connection to server. Retry?"));
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/Controller.cpp]: Format like
```
return QMessageBox::Yes == QMessageBox::question(nullptr, tr("Retry?"), tr("No connection to server. Retry?"));
```
or
```
return QMessageBox::Yes == QMessageBox::question(
	nullptr,
	tr("Retry?"),
	tr("No connection to server. Retry?"));
```

File: src/Client/Controller/Controller.cpp [hunk 4]
```diff
@@ -172,6 +203,30 @@ void Controller::OnChangePage(UIPages page)
...
+	switch(MainPage::UISettings::LANG_UI)
+	{
+	case UILangs::ENGLISH:
+		file = "en_US.qm";
+		break;
+	case UILangs::UKRAINIAN:
+		file = "uk_UA.qm";
+		break;
+	default:
+		file = "en_US.qm";
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/Controller.cpp]: Hardcoded values, consider making constants

File: src/Client/Controller/LoginPageController.cpp [hunk 5]
```diff
@@ -32,6 +37,12 @@ void LoginPageController::ConnectPage()
 		&LoginPage::GotoSignup,
 		this,
 		&LoginPageController::OnGoToSignupPage);
+
+	QObject::connect(
+		&m_page,
+		&LoginPage::LangChanged,
+		this,
+		[this](){ emit LangChanged(); });
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/LoginPageController.cpp]: Look at `connect` overloads, there should be one that connects signal to signal directly. Also you don't have to specify `QObject::connect` explicitly, unless you override connect, but you shouldn't do that anyway.

File: src/Client/View/MainPage/CategoryEditPage/CategoryEditPage.cpp [hunk 6]
```diff
@@ -109,7 +124,7 @@ void CategoryEditPage::OnRemoveClicked()
 	{
 		if(ui->Incomes->currentRow() == -1) // if user didn't select a category
 		{
-			emit ClientError("Please Select an item!");
+			emit ClientError(QApplication::tr("Please Select an item!"));
```
Comment (yaroslav-kernitskyi) [inline src/Client/View/MainPage/CategoryEditPage/CategoryEditPage.cpp]: Just `tr` doesn't work?

File: src/Client/Controller/Controller.cpp (hunk 4 above)
Comment (iodream) [reply src/Client/Controller/Controller.cpp]: replaced them with a const std::map

File: src/Client/View/MainPage/CategoryEditPage/CategoryEditPage.cpp (hunk 6 above)
Comment (iodream) [reply src/Client/View/MainPage/CategoryEditPage/CategoryEditPage.cpp]: it used to complain about context but now it's working. replaced.

**Review by yaroslav-kernitskyi — APPROVED** [review]

**Review by SantaBar6arra — APPROVED** [review]

**Pull request merged**